src/main/java/og_spipes/service/ScriptOntologyHelper.java

```diff
@@ -33,13 +33,23 @@ public Map<URI, File> moduleFile(Set<URI> modules){
             String baseURI = OntologyDao.getOntologyUri(ff);
             if(!baseURI.isBlank()){
                 String importName = baseURI + "/";
-                List<String> okModules = modules.stream().map(URI::toString)
+                List<String> okModules = modules.stream()
+                        .map(uri -> {
+                            if (uri.getFragment() != null) {
+                                return uri.getScheme() + "://" + uri.getHost() + uri.getPath() + "/" + uri.getFragment();
+                            } else {
+                                return uri.toString();
+                            }
```

Comment:
- didn't you just create normalizeUri(URI uri) method for what you are using in .map()
- what is okModules abbreviation for? consider renaming it